controllers/propagator/propagation.go

```diff
@@ -715,12 +616,10 @@ func (r *Propagator) handleRootPolicy(instance *policiesv1.Policy) error {
 
 	log.V(1).Info("Updating the root policy status")
 
-	cpcs, _ := r.calculatePerClusterStatus(instance, allDecisions, failedClusters)
-
-	// loop through all pb, update status.placement
-	sort.Slice(placements, func(i, j int) bool {
-		return placements[i].PlacementBinding < placements[j].PlacementBinding
-	})
+	cpcs, cpcsErr := r.calculatePerClusterStatus(instance, decisions)
+	if cpcsErr != nil {
+		log.Error(cpcsErr, "Failed to get at least one replicated policy")
+	}
 
 	err = r.Get(context.TODO(), types.NamespacedName{Namespace: instance.Namespace, Name: instance.Name}, instance)
 	if err != nil {
```

Comment:
I'm thinking that only after the status update should you send out the replicated policy reconcile requests. The reason being that we can run into race conditions. For example, if the replicated policy status causes the rootpolicystatus reconciler to update, the status update will fail and cause the reconcile to be restarted.

This could easily happen if the channel's buffer is full and the `handleRootPolicy` method is blocked waiting for additional space.

Reply:
Actually, I see that there is locking that should prevent the scenario I mentioned. I still think it'd be better to only send the reconcile requests after all the work is done to not duplicate reconcile requests to the replicated policy controller in the event there is an error later on that causes the root policy reconcile request to be requeued.

Reply:
Adjusting the order of this seems like a good idea.

Reply:
Changing this exposed that this `cpcsErr` will be non-nil if a replicated policy's lookup fails, which could happen quite often now that their creation is handled asynchronously. So I think it will now just be an error to ignore.